otherlibs/systhreads/st_stubs.c

```diff
@@ -948,3 +965,29 @@ static st_retcode caml_threadstatus_wait (value wrapper)
 
   CAMLreturnT(st_retcode, retcode);
 }
+
+/* Set the current thread's name. */
+CAMLprim value caml_current_thread_set_name(value name)
+{
+#if defined(_WIN32)
+  char_os *thread_name = caml_stat_strdup_to_os(String_val(name));
+  SetThreadDescription(GetCurrentThread(), thread_name);
+  caml_stat_free(thread_name);
```

Comment:
Since 5.3/trunk is currently using winpthreads on both MinGW-w64 and MSVC ports, maybe this should go through winpthreads rather than the WinAPI directly. winpthreads exposes

```c
WINPTHREAD_API pthread_t pthread_self(void);
WINPTHREAD_API int       pthread_setname_np(pthread_t thread, const char *name);
```

Reply:
I've done a little research and winpthreads doesn't use `SetThreadDescription` because it's too recent. It keeps a local copy of the thread name and throws a particular exception when a debugger is attached. Doing _both_ would cover it all.
Cf [Set a thread name in C/C++](https://learn.microsoft.com/en-us/visualstudio/debugger/tips-for-debugging-threads?view=vs-2022&tabs=csharp#set-a-thread-name-in-cc). I could send a patch to winpthreads…

Reply:
I'll do both!

Reply:
I've [sent a patch](https://sourceforge.net/p/mingw-w64/mailman/message/58829419/) to winpthreads to make it use `SetThreadDescription` in `pthread_setname_np`.

Reply:
The patch's been merged!
https://github.com/mingw-w64/mingw-w64/commit/04ec40f5bcdd3966c1bfac7e64a8bacfdc9671bd